src/main/java/com/opentable/db/postgres/embedded/PreparedDbProvider.java

```diff
@@ -90,15 +114,11 @@ public String createDatabase() throws SQLException
      */
     public DataSource createDataSource() throws SQLException
     {
-        final DbInfo db = dbPreparer.getNextDb();
-        final PGSimpleDataSource ds = new PGSimpleDataSource();
-        ds.setPortNumber(db.port);
-        ds.setDatabaseName(db.dbName);
-        ds.setUser(db.user);
-        return ds;
+        final DbInfo db = createNewDatabase();
+        return createDataSourceFromDBInfo(db);
     }
 
-    private String getJdbcUri(DbInfo db)
+    String getJdbcUri(DbInfo db)
```

Comment:
Just package-private?

Reply:
Oh, maybe just for testing?

Reply:
Right, will annotate.

Reply:
Actually, given it's OSS I  like it lean and mean and it doesn't use Guava right now.

Reply:
Right on.